cachehash/cachehash.go

```diff
@@ -14,13 +14,17 @@
 
 package cachehash
 
-import "container/list"
+import (
+	"container/list"
+	"sync"
+)
 
 type CacheHash struct {
 	h       map[interface{}]*list.Element
 	l       *list.List
 	len     int
 	maxLen  int
+	mutex   *sync.Mutex
```

Comment:
This should be an anonymous struct, which will automatically add the `Lock()` and `Unlock()` methods to your object.

Reply:
https://talks.golang.org/2012/10things.slide#3